Unable to compile ec.rs: "error[E0425]: cannot find function `_mm_*` in this scope"

Trying to compile rav1e in media-autobuild_suite (MSYS2/MinGW) for both Win32 and Win64 targets, fails already in the Win32 pass.

Brief crop:
```
error[E0425]: cannot find function `_mm_and_si128` in this scope
   --> src\ec.rs:399:66
    |
399 |       let out_v = _mm_or_si128(_mm_andnot_si128(mask, shrunk_v), _mm_and_si128(mask, expanded_v));
    |                                                                  ^^^^^^^^^^^^^ not found in this scope
help: possible candidates are found in other modules, you can import them into scope
    |
13  | use core::arch::x86::_mm_and_si128;
    |
13  | use std::arch::x86::_mm_and_si128;
    |

error[E0425]: cannot find function `_mm_storel_epi64` in this scope
   --> src\ec.rs:400:7
    |
400 |       _mm_storel_epi64(cdf.as_mut_ptr() as *mut __m128i, out_v);
    |       ^^^^^^^^^^^^^^^^ not found in this scope
help: possible candidates are found in other modules, you can import them into scope
    |
13  | use core::arch::x86::_mm_storel_epi64;
    |
13  | use std::arch::x86::_mm_storel_epi64;
    |

error[E0412]: cannot find type `__m128i` in this scope
   --> src\ec.rs:400:49
    |
400 |       _mm_storel_epi64(cdf.as_mut_ptr() as *mut __m128i, out_v);
    |                                                 ^^^^^^^ not found in this scope
help: possible candidates are found in other modules, you can import them into scope
    |
13  | use core::arch::x86::__m128i;
    |
13  | use std::arch::x86::__m128i;
    |

error: aborting due to 18 previous errors

Some errors occurred: E0412, E0425.
```

Complete log files: [logs.zip](https://0x0.st/zP5l.zip)

--- a/src/ec.rs
+++ b/src/ec.rs
@@ -374,7 +374,7 @@ impl<S> WriterBase<S> {
     }
   }
 
-  #[cfg(any(target_arch = "x86", target_arch = "x86_64"))]
+  #[cfg(target_arch = "x86_64")]
   pub fn update_cdf_4_sse2(cdf: &mut [u16], val: u32) {
     #[cfg(target_arch = "x86_64")]
     use std::arch::x86_64::*;

